Ktranslate panicing on a Device's SNMP v2c credentials check

I have over 700 devices on this single instance of ktranslate and over 7000 devices in total split across other ktranslate containers. On one of my device in the 700 I am noticing that when ktranslate tries to authenticate with the v2c credentials it panics and then the container restarts. This happens on the same device every time and the logs only seem to provide the panic message.

Can we make it so that when it hits a device like this that instead of panicing that it gracefully moves on to a new device in the list?

```2024-10-16T20:18:53.169 ktranslate/Stores53-77-snmp [Info] KTranslate Discovering SNMP devices on x.x.x.x/32.
2024-10-16T20:18:53.188 ktranslate/Stores53-77-snmp [Info] KTranslate Starting to check 1 ips in x.x.x.x/32
2024-10-16T20:18:53.189 ktranslate/Stores53-77-snmp [Info] KTranslate 1/1) Host found at x.x.x.x, Manufacturer: , Name:  -- now attempting checking snmp connectivity
2024-10-16T20:18:53.189 ktranslate/Stores53-77-snmp [Info] KTranslate 1/1) Running with SNMP v3: Priv: DES Auth: SHA
2024-10-16T20:18:53.232 ktranslate/Stores53-77-snmp [Warn] KTranslate Cannot get device metadata on x.x.x.x: unknown username. Check for correct snmp credentials.
2024-10-16T20:18:53.233 ktranslate/Stores53-77-snmp [Info] KTranslate 1/1) Running with SNMP v3: Priv: AES Auth: SHA
2024-10-16T20:18:53.291 ktranslate/Stores53-77-snmp [Warn] KTranslate Cannot get device metadata on x.x.x.x: unknown username. Check for correct snmp credentials.
2024-10-16T20:18:53.291 ktranslate/Stores53-77-snmp [Info] KTranslate 1/1) Running with SNMP v3: Priv: AES Auth: MD5
2024-10-16T20:18:53.354 ktranslate/Stores53-77-snmp [Warn] KTranslate Cannot get device metadata on x.x.x.x: unknown username. Check for correct snmp credentials.
2024-10-16T20:18:53.354 ktranslate/Stores53-77-snmp [Info] KTranslate 1/1) Running with SNMP v2c
panic: interface conversion: interface {} is []uint8, not string

--- a/hack/grafana-cloud/docker-compose.yml
+++ b/hack/grafana-cloud/docker-compose.yml
@@ -14,6 +14,9 @@ services:
       - alloy
     command:
       - --format=otel
+      - --sinks=otel
+      - --metrics=jchf
+      - --tee_logs=true
       - --otel.protocol=grpc
       - --otel.endpoint=http://alloy:4317/
       - --snmp=/snmp.yml

--- a/pkg/inputs/snmp/disco.go
+++ b/pkg/inputs/snmp/disco.go
@@ -196,6 +196,9 @@ func doubleCheckHost(result scan.Result, timeout time.Duration, ctl chan bool, m
 	// Get the token to allow us to run.
 	_ = <-ctl
 	defer func() {
+		if r := recover(); r != nil {
+			log.Errorf("Recovered from panic, skipping device: %v", r)
+		}
 		wg.Done()
 		ctl <- true
 	}()

--- a/pkg/inputs/snmp/metadata/device_metadata.go
+++ b/pkg/inputs/snmp/metadata/device_metadata.go
@@ -164,7 +164,11 @@ func (dm *DeviceMetadata) poll(ctx context.Context, server *gosnmp.GoSNMP) (*kt.
 		case SNMP_sysDescr:
 			md.SysDescr = string(value.([]byte))
 		case SNMP_sysObjectID:
-			md.SysObjectID = value.(string)
+			if s, ok := value.(string); ok {
+				md.SysObjectID = s
+			} else if s, ok := value.([]byte); ok {
+				md.SysObjectID = string(s)
+			}
 		case SNMP_sysContact:
 			md.SysContact = string(value.([]byte))
 		case SNMP_sysName:
@@ -315,7 +319,11 @@ func GetBasicDeviceMetadata(log logger.ContextL, server *gosnmp.GoSNMP) (*kt.Dev
 		case SNMP_sysDescr:
 			md.SysDescr = string(value.([]byte))
 		case SNMP_sysObjectID:
-			md.SysObjectID = value.(string)
+			if s, ok := value.(string); ok {
+				md.SysObjectID = s
+			} else if s, ok := value.([]byte); ok {
+				md.SysObjectID = string(s)
+			}
 		case SNMP_sysContact:
 			md.SysContact = string(value.([]byte))
 		case SNMP_sysName:

